### ISSUE TITLE
mismatch between unprocess and process

### ISSUE THREAD

**Issue by xlying55** [timothybrooks/unprocessing#2]
Hi, nice work. I meet a question, that I want to denoise in the sRGB space, and I also wish to add noise in the RAW space. However, during testing, even if I do not add any noise, I just use unprocess.py and process.py to convert an image to RAW space and convert back, the results are different to the original. Can you help me with that? The first image is the input, the second is using unprocess.py then use process.py. 
Thanks 
![01_gt](https://user-images.githubusercontent.com/40621671/100359705-e09fd000-3032-11eb-8738-fd1abf5c92b0.png)
![processed0](https://user-images.githubusercontent.com/40621671/100359711-e2699380-3032-11eb-9d5a-225e4ccf7652.png)

**Comment by timothybrooks** [comment]
Hi -- thanks for the question. Unprocessing is a method for creating plausible raw images, which improves the realism of training data. However it involves lossy steps, such as subsampling to create a Bayer mosaic. Therefore the original sRGB image cannot be perfectly reconstructed from an unprocessed image. This is why we forward process the unprocessed images with and without noise to attain input and label data. Please refer to the paper, especially Figure 2, where the Noise-free sRGB image used for computing loss matches the processing of the denoised image, but slightly differs from the original input image. For your use case you may want to process both noisy and noise free images like we did.

**Comment by IceClear** [comment]
@timothybrooks Hi, thanks for the wonderful work. Actually, I found producting the 'rgb_gain' in the 'apply_gains' function in 'process.py' can significantly correct the color mismatching problem. I wonder why the 'rgb_gain' is ignored during the white balance process. I think it should be considered since it has been applied during the inverse process: 'safe_invert_gains' in the 'unprocess.py'. Thanks.

**Comment by arnaud-nt2i** [comment]
@swaggyijk @IceClear @timothybrooks 
adding rgb_gains to  Process/apply_gains give a little improvement.
But the main problem is that smoothstep (tone maping) is missing in process (after gamma):
`def smoothstep(image):
  image = tf.clip_by_value(image, 0.0, 1.0)
  return 3.0 * image ** 2 - 2 * image ** 3`
  
 Using a better demosaic algo does help as well [https://github.com/Berezniker/CV_Bayer](url)
